### PR TITLE
AQ-71 routing

### PR DESCRIPTION
**Pull request by ToJestKrzysio**
# Description
Updated routes, and auth provider.

issue [AQ-71](https://azeno.youtrack.cloud/issue/AQ-71/React-login-registration-startup-game-routes)

## Type of change

Please delete options that are not relevant.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?
Added new tests to cover updated auth provider.
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The unrelated memme has been added

image

### CODE REVIEW

File: quiz/web/src/router.tsx [hunk 1]
```diff
@@ -1,13 +1,26 @@
+import { UserRequired } from "modules/common/components";
 import { createBrowserRouter } from "react-router-dom";
 
 // eslint-disable-next-line import/prefer-default-export
 export const router = createBrowserRouter([
     {
         path: "/",
-        element: <h1>Home</h1>,
+        element: (
```
Comment (martynawitkowska) [inline quiz/web/src/router.tsx:8]: no `/game/` endpoint ? 🥺

Comment (ToJestKrzysio) [reply quiz/web/src/router.tsx:8]: I've assumed home to be the game endpoint. If you start an app http://azeno.com/ you should go to select game mode (home).

Comment (ToJestKrzysio) [reply quiz/web/src/router.tsx:8]: if you don't have an account `<UserRequired>` will redirect you to start.

**Comment by ToJestKrzysio** [comment]
Changes required to adhere to the application flow better.

**Review by martynawitkowska — APPROVED** [review]
So endpoints are there, they all have the same form, but some stuff is changing when one enters the endpoint, so I l'd say everything works😊

File: quiz/web/.eslintrc.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
                 "extensions": [".tsx", ".ts"]
             }
         ],
+        "import/no-extraneous-dependencies": ["error", {"devDependencies": true}],
```
Comment (ToJestKrzysio) [inline quiz/web/.eslintrc.json:35]: Will stop eslint from throwing errors about packages not present in dependencies beacause they are in devDeps.

**Pull request merged**